core/lib/multivm/src/versions/vm_latest/types/internals/transaction_data.rs

```diff
@@ -62,6 +62,12 @@ impl From<Transaction> for TransactionData {
                     U256::zero()
                 };
 
+                let should_deploy_contract = if execute_tx.execute.contract_address.is_none() {
```

Comment:
Looks somewhat dangerous. Especially if we will allow to use `None` as `to` even without EVM simulator

Reply:
Fixed by asserting that the EVM simulator is enabled before setting the field; PTAL.